dynawo/sources/Models/Modelica/Dynawo/Electrical/Controls/Transformers/BaseClasses.mo

```diff
@@ -98,142 +91,77 @@ partial model BaseTapChangerPhaseShifter_MAX "Base model for tap-changers and ph
+    valueAboveMax = pre(valueAboveMax);
+    tValueAboveMaxWhileRunning = Constants.inf;
+    valueUnderStop = pre(valueUnderStop);
   end when;
 
   lookingToDecreaseTap = running.value and valueAboveMax and decreaseTapToDecreaseValue;
   lookingToIncreaseTap = running.value and valueAboveMax and increaseTapToDecreaseValue;
 
-  //The "not(locked)" condition is important to prevent unwanted transitions when the phase-shifter leaves the "locked" state
-  when running.value and valueAboveMax and not(locked) then
-    tValueAboveMaxWhileRunning= time;
-  elsewhen running.value and valueUnderStop and not(locked) then
-    tValueAboveMaxWhileRunning= Constants.inf;
-  elsewhen running.value and locked then
-    tValueAboveMaxWhileRunning= Constants.inf;
-  end when;
-
-  //Transition to "locked"
+  //Transition to "Locked" (possible from any state and prioritary)
   when (not running.value) or locked then
     state = State.Locked;
     tap.value = pre(tap.value);
     tTapUp = Constants.inf;
     tTapDown = Constants.inf;
-  //Transition from "locked" to "standard"
-  elsewhen running.value and not(locked) and not(valueAboveMax) and pre(state) == State.Locked then
-    state = State.Standard;
-    tap.value = pre(tap.value);
-    tTapUp = Constants.inf;
-    tTapDown = Constants.inf;
-  //Transition from "locked" to "waitingToMoveDown"
-  elsewhen running.value and not(locked) and lookingToDecreaseTap and pre(state) == State.Locked then
+  //Transition to "WaitingToMoveDown" (possible from any state except down states)
+  elsewhen lookingToDecreaseTap and (pre(state) == State.Standard or pre(state) == State.MoveUp1 or pre(state) == State.MoveUpN or pre(state) == State.WaitingToMoveUp or pre(state) == State.Locked) then
     state = State.WaitingToMoveDown;
     tap.value = pre(tap.value);
     tTapUp = Constants.inf;
     tTapDown = Constants.inf;
-  //Transition from "locked" to "waitingToMoveUp"
-  elsewhen running.value and not(locked) and lookingToIncreaseTap and pre(state) == State.Locked then
+  //Transition to "WaitingToMoveUp" (possible from any state except up states)
+  elsewhen lookingToIncreaseTap and (pre(state) == State.Standard or pre(state) == State.MoveDown1 or pre(state) == State.MoveDownN or pre(state) == State.WaitingToMoveDown or pre(state) == State.Locked) then
     state = State.WaitingToMoveUp;
     tap.value = pre(tap.value);
     tTapUp = Constants.inf;
     tTapDown = Constants.inf;
-  //Transition from "standard" to "waitingToMoveUp"
-  elsewhen running.value and lookingToIncreaseTap and pre(state) == State.Standard then
-    state = State.WaitingToMoveUp;
-    tap.value = pre(tap.value);
-    tTapUp = pre(tTapUp);
-    tTapDown = pre(tTapDown);
-  //Transition from "waitingToMoveUp" to "standard"
-  elsewhen running.value and valueUnderStop and pre(state) == State.WaitingToMoveUp then
+  //Transition to "Standard" (possible from any state)
+  elsewhen not(valueAboveMax) and pre(state) <> State.Standard then
```

Comment:
in the original model, for the transitions to standard, we sometimes have not(valueAvoveMax) and sometimes valueUnderStop, are you sure that it is equivalent to this new implementation? The phaseShifterI stops regulating when I<IStop, not when I<IMax...

Reply:
It seems that's a mistake -> I will correct it.

Reply:
Corrected.